openhcl/fixed_pool_alloc/src/lib.rs

```diff
@@ -0,0 +1,530 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! This module implements a fixed memory allocator for allocating pages at specific location.
+
+#![cfg(target_os = "linux")]
+#![warn(missing_docs)]
+
+mod mapped_dma;
+
+pub use mapped_dma::FixedDmaBuffer;
+
+use anyhow::Context;
+use hvdef::HV_PAGE_SIZE;
+use inspect::Inspect;
+use memory_range::MemoryRange;
+use parking_lot::Mutex;
+use std::num::NonZeroU64;
+use std::sync::Arc;
+use thiserror::Error;
+use user_driver::memory::save_restore::MemPoolSavedState;
+use user_driver::memory::save_restore::MemPoolState;
+use user_driver::memory::MemoryBlock;
+use user_driver::vfio::VfioDmaBuffer;
+use user_driver::HostDmaAllocator;
+
+/// Error returned when unable to allocate memory.
+#[derive(Debug, Error)]
```

Comment:
I am using Error because it was easy to integrate, also will be superseded by page_pool change. So leaving it as is but keeping in mind for future improvements.

Reply:
To be superseded by PagePool allocator.